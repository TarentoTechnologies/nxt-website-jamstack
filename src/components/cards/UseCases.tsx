import React from "react";

import { useCasesSubTitle } from "./UseCases.module.css";

interface UseCasesProps {
  title: string;
  imgSrc: string;
}

export const UseCases = ({ title, imgSrc }: UseCasesProps) => {
  return (
    <div className="layoutDetails d-feature useCases" data-testid="use-cases">
      <div className={`col-md-4 col-sm-6 col-xs-12 d-flex align-items-center`}>
        <div className="">
          <img src={imgSrc} alt={title} title={title} />
        </div>
        <div className="">
          <div className={useCasesSubTitle}>{title}</div>
        </div>
      </div>
    </div>
  );
};
