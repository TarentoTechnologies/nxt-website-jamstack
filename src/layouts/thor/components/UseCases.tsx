import { useCasesSubTitle } from "../Thor.module.css";

interface CaseStudiesProps {
  data: {
    title?: string;
    list?: { imgSrc: string; imgText: string }[];
    subText?: string;
  };
}

export const UseCases = ({ data }: CaseStudiesProps) => {
  return (
    <div className="layout layoutBg">
      <div className={`container-fluid d-flex justify-content-center`}>
        <div className={`containerService paddingLeftRight15`}>
          <h1 className="pb-5">{data.title}</h1>
          <div className="row layoutDetails d-feature useCases">
            {data.list?.map((item, index) => (
              <div
                className={`col-lg-4 mb-5 col-sm-6 col-xs-12 d-flex align-items-center`}
                key={index}
              >
                <div className="pull-left wow fadeInUp  animated">
                  <img
                    src={item.imgSrc}
                    alt={item.imgText}
                    title={item.imgText}
                  />
                </div>
                <div className="wow fadeInUp  animated">
                  <div className={useCasesSubTitle}>{item.imgText}</div>
                </div>
              </div>
            ))}
            <div className={`${useCasesSubTitle} mt-3 ms-0`}>
              {data.subText}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
