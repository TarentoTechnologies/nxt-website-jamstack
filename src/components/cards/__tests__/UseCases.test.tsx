import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { UseCases } from "../UseCases";

describe("Test if the UseCases component is working as expected", () => {
  it("<UseCases /> matches snapshot", () => {
    const component = render(
      <UseCases
        title={"Usecase1"}
        imgSrc={
          "https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_2f3dc77aa7.png"
        }
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<UseCases /> renders without exploding", () => {
    const component = render(
      <UseCases
        title={"Usecase1"}
        imgSrc={
          "https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_2f3dc77aa7.png"
        }
      />
    );
    expect(component.getByTestId("use-cases")).toBeInTheDocument();
  });
});
