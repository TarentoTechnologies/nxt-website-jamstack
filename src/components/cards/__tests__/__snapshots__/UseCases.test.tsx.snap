// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the UseCases component is working as expected <UseCases /> matches snapshot 1`] = `
<div>
  <div
    class="layoutDetails d-feature useCases"
    data-testid="use-cases"
  >
    <div
      class="col-md-4 col-sm-6 col-xs-12 d-flex align-items-center"
    >
      <div
        class=""
      >
        <img
          alt="Usecase1"
          src="https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_2f3dc77aa7.png"
          title="Usecase1"
        />
      </div>
      <div
        class=""
      >
        <div
          class="useCasesSubTitle"
        >
          Usecase1
        </div>
      </div>
    </div>
  </div>
</div>
`;
